Makefile

```diff
@@ -150,7 +156,11 @@ $(FILTERLIB_OUTPUT): $(FILTERLIB_OBJS)
 	$(AR) qc $@ $^
 	$(RANLIB) $@
 
-test: test/test rust-test
+HTTPTESTSERVER_RUN_LOCATION=test/http-test-server
+$(HTTPTESTSERVER_RUN_LOCATION): $(HTTPTESTSEVER_OUTPUT)
```

Comment:
The `check` and `ci-check` targets already depend on the `test` target which makes sure that the test binaries `test/test` (C++), <???> (Rust) and http test utility (`http-test-server`) are build.
Afterwards, the `test/test` (C++) binary will start the `http-test-server` by itself, so that needs no further involvement from the Makefile.

However, the C++ test binary needs to know where to find the `http-test-server` binary.
That location depends on a few variables which are only known at build time (`CARGO_TARGET_DIR`, `CARGO_BUILD_TARGET`, `BUILD_TYPE`)
To avoid that dependency, I decided to always copy the `http-test-server` binary to a known location (`test/http-test-server`)